### PR TITLE
Adds parsing of `ShellTask` and `IconTask` from the config file

### PR DESCRIPTION
**Pull request by agoscinski**
The parameters specified in the config file corresponding to specific task plugins file are now passed to their corresponding classes in core. ShellTask and IconTask. For that pydantic discrimantors are used. See for more information
https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions-with-callable-discriminator

To test the new changes formatting of represented member variables has been added to the formatting of `core.Task` objects.

Documentation to the `core.Plugin` has been added and the class has been renamed to `core.TaskPlugin`.

### CODE REVIEW

File: src/sirocco/core/graph_items.py [hunk 1]
```diff
@@ -20,31 +20,35 @@ class GraphItem:
...
 
 
-class Plugin(type):
+class TaskPlugin(type):
     """Metaclass for plugin tasks inheriting from Task
 
     Used to register all plugin task classes"""
 
-    classes: dict[str, type] | None = None
+    classes: ClassVar[dict[str, type[Task]]] = {}
```
Comment (agoscinski) [inline src/sirocco/core/graph_items.py:28]:
```suggestion
    classes: ClassVar[dict[str, type[Task]]] = {}  # is valid because is singleton
```

File: src/sirocco/core/graph_items.py [hunk 2]
```diff
@@ -20,31 +20,35 @@ class GraphItem:
     coordinates: dict = field(default_factory=dict)
 
 
-class Plugin(type):
+class TaskPlugin(type):
```
Comment (agoscinski) [inline src/sirocco/core/graph_items.py:23]: renamed to TaskPlugin (can undo this change), just added doc in this class and set to empty dictionary the classes in the class definition

File: src/sirocco/core/graph_items.py [hunk 3]
```diff
@@ -20,31 +20,35 @@ class GraphItem:
...
 
 
 @dataclass
-class Task(GraphItem, metaclass=Plugin):
+class Task(GraphItem, metaclass=TaskPlugin):
     """Internal representation of a task node"""
 
     plugin: ClassVar[str] = "_BASE_TASK"
-    color: ClassVar[str] = "light_red"
+    color: ClassVar[str] = field(default="light_red", repr=False)
```
Comment (agoscinski) [inline src/sirocco/core/graph_items.py:51]: repr is False no it is not shown in formatting

File: src/sirocco/parsing/_yaml_data_models.py [hunk 4]
```diff
@@ -3,13 +3,16 @@
...
 from pathlib import Path
-from typing import Any
+from typing import Annotated, Any, Literal
 
 from isoduration import parse_duration
 from isoduration.types import Duration  # pydantic needs type # noqa: TCH002
-from pydantic import BaseModel, ConfigDict, Field, field_validator, model_validator
+from pydantic import BaseModel, ConfigDict, Discriminator, Field, Tag, field_validator, model_validator
 
-from ._utils import TimeUtils
```
Comment (agoscinski) [inline src/sirocco/parsing/_yaml_data_models.py:12]: I think absolute import path are more readable

File: src/sirocco/parsing/_yaml_data_models.py [hunk 5]
```diff
@@ -29,18 +32,30 @@ class _NamedBaseModel(BaseModel):
     name: str
 
     def __init__(self, /, **data):
+        super().__init__(**self.merge_name_and_specs(data))
+
+    @staticmethod
+    def merge_name_and_specs(data: dict) -> dict:
```
Comment (agoscinski) [inline src/sirocco/parsing/_yaml_data_models.py:38]: was useful for reusage

File: src/sirocco/parsing/_yaml_data_models.py [hunk 6]
```diff
@@ -298,6 +324,25 @@ class ConfigData(BaseModel):
     generated: list[ConfigGeneratedData] = []
 
 
+def get_plugin_from_named_base_model(data: dict) -> str:
+    name_and_specs = _NamedBaseModel.merge_name_and_specs(data)
+    if name_and_specs.get("name", None) == "ROOT":
+        return Task.plugin
+    plugin = name_and_specs.get("plugin", None)
+    if plugin is None:
+        msg = "Could not find plugin name in {data}"
```
Comment (agoscinski) [inline src/sirocco/parsing/_yaml_data_models.py:333]:
```suggestion
        msg = f"Could not find plugin name in {data}"
```

File: src/sirocco/pretty_print.py [hunk 7]
```diff
@@ -66,13 +66,14 @@ def as_item(self, content: str) -> str:
...
-    def format(self, obj: Any):
+    def format(self, obj: Any) -> str:
         """
         Dispatch formatting based on node type.
 
         Default implementation simply calls str()
         """
-        return str(obj)
+        # this prevents empty strings being not visibliy displayed
+        return repr(obj) if isinstance(obj, str) else str(obj)
```
Comment (agoscinski) [inline src/sirocco/pretty_print.py:76]: for example if the string is `command_option = ''` it was formatted to `comand_option:  `. With this change it is formatted to `comand_option: ''`

File: tests/files/data/test_config_small.txt [hunk 8]
```diff
@@ -21,8 +33,20 @@ cycles:
...
+            command: '$PWD/tests/files/scripts/icon.py'
+            command option: ''
+            input arg options: {'icon_restart': '--restart'}
   - lastly:
       tasks:
         - cleanup:
             wait on:
-              - icon [date: 2026-05-01 00:00:00]
\ No newline at end of file
+              - icon [date: 2026-05-01 00:00:00]
```
Comment (agoscinski) [inline tests/files/data/test_config_small.txt:46]: I think this line is marked "removed" because it was the end of the line before

**Review by leclairm — APPROVED** [review]
